src/task/MonitorTask.ts

```diff
@@ -79,10 +79,30 @@ export default class MonitorTask
     this.context.realtimeController.realtimeSubscribeToLocalSignalStrengthChange(
       this.checkAndSendWeakSignalEvent
     );
-
     this.context.connectionMonitor.start();
     this.context.statsCollector.start(this.context.signalingClient, this.context.videoStreamIndex);
     this.context.signalingClient.registerObserver(this);
+    const attendeeId = this.context.meetingSessionConfiguration.credentials.attendeeId;
+
+    const startTimestampMs = Date.now();
+    this.context.realtimeController.realtimeSubscribeToAttendeeIdPresence(
+      (presentAttendeeId: string, present: boolean) => {
+        if (attendeeId === presentAttendeeId && present) {
```

Comment:
As discussed before, you should unsubscribe when you receive the first callback. Otherwise, `MonitorTask` keeps publishing an event when the attendee is dropped and reconnected to the meeting.

(If you want to keep updating `selfAttendeePresentDurationMs`, you may exclude the duration while the `drop` (4th parameter) is `true`. But this is overkill.)